Реализованы задачи из спринта

DesidiamS opened the pull request.

Commit:
Реализованы задачи из спринта

Commit:
Убраны комментарии и исправлен код-стайл

Commit:
Убраны комментарии и исправлен код-стайл

bismuthD commented on src/main/java/ru/practicum/shareit/item/mappers/ItemMapper.java:
```diff
@@ -0,0 +1,26 @@
+import ru.practicum.shareit.item.model.Item;
+import ru.practicum.shareit.user.User;
+
+public class ItemMapper {
```
для классов мапперов хорошо использовать аннотацию @Mapper , так же вместо нее можно использовать @NoArgsConstructor(access = AccessLevel.PRIVATE)

bismuthD commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,59 @@
+
+    @GetMapping
+    List<Item> getAllItems(@RequestHeader HttpHeaders headers) {
+        Long userId = Long.valueOf(Objects.requireNonNull(headers.get("X-Sharer-User-Id")).get(0));
```
X-Sharer-User-Id дублируется, следует вынести значение в константу, для этого лучше выделить служебный класс, в который будешь добавлять все константы по проекту

bismuthD commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,59 @@
+    @PostMapping
+    public Item createItem(@RequestBody @Valid ItemDto itemDto,
+                           @RequestHeader HttpHeaders headers) {
+        Long userId = Long.valueOf(Objects.requireNonNull(headers.get("X-Sharer-User-Id")).get(0));
```
так же в контроллеры следует добавить логирование

bismuthD reviewed (commented):
Отличная работа, оставил несколько комментариев на доработку

Commit:
- Добавлено логирование в контроллеры.
- Заголовок id пользователя вынесен в константы.
- В маппер добавлена аннотация @NoArgsConstructor(access = AccessLevel.PRIVATE)

Commit:
- Исправлен кодстайл
